connmgr/connmanager.go

```diff
@@ -137,14 +142,33 @@ type Config struct {
 	// to.  If nil, no new connections will be made automatically.
 	GetNewAddress func() (net.Addr, error)
 
-	// Dial connects to the address on the named network. Either Dial or
-	// DialAddr need to be specified (but not both).
+	// Dial connects to the address on the named network.
+	// Only one of Dial, DialAddr, DialContext, or DialAddrConext can and
+	// must be specified.
+	//
+	// Deprecated: Use DialContext instead.
 	Dial func(network, addr string) (net.Conn, error)
 
-	// DialAddr is an alternative to Dial which receives a full net.Addr instead
-	// of just the protocol family and address. Either DialAddr or Dial need
-	// to be specified (but not both).
+	// DialAddr is an alternative to DialContext which receives a full
+	// net.Addr instead of just the protocol family and address.
+	// Only one of Dial, DialAddr, DialContext, or DialAddrContext can and
+	// must be specified.
+	//
+	// Deprecated: Use DialAddrContext instead.
 	DialAddr func(net.Addr) (net.Conn, error)
+
+	// DialContext connects to the address on the named network using the
+	// provided context.
+	//
+	// Only one of Dial, DialAddr, DialContext, or DialAddrContext can and
+	// must be specified.
+	DialContext func(ctx context.Context, network, addr string) (net.Conn, error)
+
+	// DialAddr is an alternative to DialContext which receives a full
```

Comment:
Function name in comment doesn't match the real name.

Reply:
fixed.